spring1/src/main/java/com/unitbv/lifecycle/FunBean.java

```diff
@@ -13,4 +13,23 @@
     private Logger logger = LoggerFactory.getLogger(FunBean.class);
```

Comment:
FunBean should implement DisposableBean and InitializingBean interfaces and their methods.